src/codesyncd/handlers/tabs_handler.ts

```diff
@@ -0,0 +1,169 @@
+import path from "path";
+
+import { glob } from "glob";
+
+import { ITabYML } from "../../interface";
+import { generateSettings } from "../../settings";
+import { readYML } from "../../utils/common";
+import { getRandomIndex, getTabsBeingProcessed } from "../utils";
+import { removeFile } from "../../utils/file_utils";
+import { TAB_FILES_PER_ITERATION, TAB_SIZE_LIMIT } from "../../constants";
+import { CodeSyncLogger } from "../../logger";
+import { TabValidator } from "../validators/tab_validator";
+import { TabHandler } from "./tab_handler";
+import { ConfigUtils } from "../../utils/config_utils";
+
+export class TabsHandler {
+    // @ts-ignore
+    tabs: ITabYML[];
+    // @ts-ignore
+    accessToken: string;
+
+    settings: any;
+    configJSON: any;
+    configRepo: any;
+
+    constructor(repoTab: ITabYML[] | null = null, accessToken: string | null = null) {
```

Comment:
We shouldn't be calling constructor is we don't have repoTab, right?

Reply:
Same goes to `accessToken`

Reply:
You are right. I am considering refactoring code so as to move `getYMLFiles` out from `tabs_handler` to another file.

As we do not have access to `repoTab` in `buffer_handler` (where we call `getYMLFiles`),

I did not want to make a new file but since `tabs_handler` & `tab_handler` require constructor params, I am thinking of making a new util file for method: `getYMLFiles`.

What do you think?

Reply:
Sure, whatever looks cleaner